src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -7941,113 +7947,154 @@ instruct xorL_reg_imm(iRegLNoSp dst, iRegL src1, immLAdd src2) %{
 // ============================================================================
 // MemBar Instruction
 
-instruct load_fence() %{
+// ZTSO
+
+instruct no_membar_rvtso() %{
+  predicate(UseZtso);
   match(LoadFence);
-  ins_cost(ALU_COST);
+  match(StoreFence);
+  match(StoreStoreFence);
+  match(MemBarAcquire);
+  match(MemBarRelease);
+  match(MemBarStoreStore);
+  match(MemBarAcquireLock);
+  match(MemBarReleaseLock);
 
-  format %{ "#@load_fence" %}
+  ins_cost(0);
+
+  size(0);
 
+  format %{ "no_membar_rvtso elided/tso (empty encoding)" %}
   ins_encode %{
-    __ membar(MacroAssembler::LoadLoad | MacroAssembler::LoadStore);
+    __ block_comment("no_membar_rvtso");
```

Comment:
And here: s/no_membar_rvtso/unnecessary_membar_rvtso/

Reply:
Fixed